sdk/overview.md

```diff
@@ -0,0 +1,124 @@
+# Abstract
+
+This document outlines a proposal for mobile service SDK 
+
+# Goals
+
+Main goal for this document is to provide full overview of the future mobile SDK.
+Document outlines a proposal for mobile service SDK's and all related elements.
+Proposal basing on industry standards for building multi service SDK's. 
+
+## Primary Goals
+
+Provide top level architecture and delivery model for mobile SDK's.
+
+## Non Goals
+
+Describe individual service SDK's:
+This will be done in separate proposals.
+
+# Problem Description
+
+Selecting best architecture used to build mobile SDK . 
+
+## Elements to include
+
+1. Git hosting
+
+How to host SDK's in git.
+Single repository for platform or service etc.?
+
+2. SDK distribution to end users
+Platform specific way of distributing packages
+Can be maven, cocoapods etc. 
+
+3. Documentation for SDK
+Api documentation and top level SDK documentation. 
+
+4. Architecture
+How to allow users to interact with SDK.
+Allow flexibility on network layer etc.
+How to prevent users from including large binary when they need single service.
+ 
+5. Adding common services and addons for all services
+Functionalities like network handling, logging, monitoring and audit, error handling, analytics.
+
+6. Developer tooling 
+How to build or provide (plugins, extensions) to service
+
+7. Size constraints 
+For example method limit on android
+
+8. Platform language
+Supporting multiple flavours (languages per platform) - Swift vs ObjectiveC. Java vs Kotlin etc.
+
+# Proposed Solution
+
+Modular SDK For SDK that can contain multiple non related services 
+
+## Overview
+
+Modular architecture is a software design technique that emphasizes separating the functionality of a program into independent, interchangeable modules, such that each contains everything necessary to execute only one aspect of the desired functionality.
+
+Modular architecture is well known as practice for the server side development, but it can be also applied when building mobile SDK. In modular mobile SDK, code can be divided into separate packages that can be used separately by specific platform package manager. Each package depends on core package that needs to be added as separate dependency.
```

Comment:
Is this similar to how Angular 2 does things now? It sounds very similar

Reply:
This aligns with how Google Firebase, Amazon AWS or many other companies split their SDK's now. Angular2 is also very similar, but it has much more modules, with core being lightweight and separate http modules. To avoid overhead here we assume that networking logging etc. will be in core.